### PR TITLE
Simplify do_next().

### PR DESCRIPTION
**Pull request by vext01**
The function `do_next()` is long and hard to follow.

These changes simplify it by pulling code out into functions.

No functional change.

### CODE REVIEW

File: hwtracer/src/decode/ykpt/mod.rs [hunk 1]
```diff
@@ -294,6 +294,116 @@ impl<'t> YkPTBlockIterator<'t> {
...
+    /// Follow the successor of the block described by the blockmap entry `ent`.
+    fn follow_blockmap_successor(&mut self, ent: &BlockMapEntry) -> Result<Block, HWTracerError> {
+        match ent.successor() {
+            SuccessorKind::Unconditional { target } => {
+                if let Some(target_off) = target {
+                    self.cur_loc = ObjLoc::MainObj(*target_off);
+                    return self.lookup_block_from_main_bin_offset(*target_off);
+                } else {
+                    // Divergent control flow.
+                    todo!();
```
[the remark this answers is not in the document]
Comment (vext01) [reply hwtracer/src/decode/ykpt/mod.rs:338]: I admit I had to look this up in the compiler code.

There are cases where, in the eyes of LLVM, a block is technically considered to "fallthru" even though there is no block to fallthru to. In these cases, a prior call must have diverged (i.e. it calls something which doesn't return).

So, the end such blocks isn't reachable, if that's what you mean.

Or did you mean is the `todo!()` reachable? It is entirely possible to reach the `todo()`, but I don't want to think about this case just now, so I'm letting it crash.

[the remark this answers is not in the document]
Comment (ltratt) [reply hwtracer/src/decode/ykpt/mod.rs:338]: In Rust `todo!()` means "we haven't finished this code yet" but `unreachable!()` means "we should never ever get here". I'm not sure which this is supposed to be.

[the remark this answers is not in the document]
Comment (vext01) [reply hwtracer/src/decode/ykpt/mod.rs:338]: It should be a `todo!()`. It is reachable.

The unreachability I'm talking about about is at the LLVM IR level.

[the remark this answers is not in the document]
Comment (ltratt) [reply hwtracer/src/decode/ykpt/mod.rs:338]: `unreachable` should be thought of as more "this is an assert that we shouldn't have got here". Dunno if that helps clarify?

[the remark this answers is not in the document]
Comment (vext01) [reply hwtracer/src/decode/ykpt/mod.rs:338]: I know the difference :+1: 

This is definitely a `todo!`.

[the remark this answers is not in the document]
Comment (ltratt) [reply hwtracer/src/decode/ykpt/mod.rs:338]: Got it!
